### PR TITLE
[PM-10977] Don't show welcome carousel when in an extension

### PR DESCRIPTION
**Pull request by KatherineInCode**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-10977

## 📔 Objective

This prevents the welcome carousel from being displayed when using the app as an extension (e.g. for autofill).

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: BitwardenShared/UI/Auth/AuthRouter.swift [hunk 1]
```diff
@@ -22,7 +25,11 @@ final class AuthRouter: NSObject, Router {
     /// - Parameter services: The services used by this router.
     ///
     /// - Parameters:
-    init(services: Services) {
+    init(
+        isInAppExtension: Bool,
+        services: Services
```
Comment (matt-livefront) [inline BitwardenShared/UI/Auth/AuthRouter.swift:30]: ⛏️ Can you add `isInAppExtension` to the docs here?

**Review by matt-livefront — APPROVED** [review]
Looks good!

File: BitwardenShared/UI/Auth/Extensions/AuthRouter+Redirects.swift [hunk 2]
```diff
@@ -208,7 +208,8 @@ extension AuthRouter {
             // If no account can be set to active, go to the landing or carousel screen.
             let isCarouselEnabled: Bool = await services.configService.getFeatureFlag(.nativeCarouselFlow)
             let introCarouselShown = await services.stateService.getIntroCarouselShown()
-            return isCarouselEnabled && !introCarouselShown ? .introCarousel : .landing
+            let shouldShowCarousel = isCarouselEnabled && !introCarouselShown
+            return shouldShowCarousel && !isInAppExtension ? .introCarousel : .landing
```
Comment (fedemkr) [inline BitwardenShared/UI/Auth/Extensions/AuthRouter+Redirects.swift:212]: ⛏️ I believe we could move the condition check into the `shouldShowCarousel` line given that is one of the conditions for that to happen so the last line is simpler and all the conditions are in one place.
```suggestion
            let shouldShowCarousel = isCarouselEnabled && !introCarouselShown && !isInAppExtension
            return shouldShowCarousel ? .introCarousel : .landing
```

**Review by fedemkr — APPROVED** [review]
LGTM, just one small thing but I'll leave it to you whether you think it's worth to change it 😄

File: BitwardenShared/UI/Auth/Extensions/AuthRouter+Redirects.swift (hunk 2 above)
Comment (KatherineInCode) [reply BitwardenShared/UI/Auth/Extensions/AuthRouter+Redirects.swift:212]: 🤔 I kind of feel like it's six of one, half dozen of the other. But putting all the conditions in one line is reasonable.

**Review by fedemkr — APPROVED** [review]

**Review by phil-livefront — APPROVED** [review]

**Pull request merged**